crates/oxide/src/scanner/mod.rs

```diff
@@ -402,6 +425,44 @@ fn read_all_files(changed_content: Vec<ChangedContent>) -> Vec<Vec<u8>> {
         .collect()
 }
 
+#[tracing::instrument(skip_all)]
+fn extract_css_variables(blobs: Vec<Vec<u8>>) -> Vec<String> {
```

Comment:
I started passing options to the other `parse_all_blobs` function but it looked a bit messy. So duplicated it instead (even though there is a good chunk of duplication going on).

Reply:
I feel like the only difference really should be that we no-op the `Candidate` machine though? Can you elaborate what you meant with messy?

Reply:
I started passing through the contents and the extension and it leaked everywhere. Can see that in this commit: https://github.com/tailwindlabs/tailwindcss/pull/17433/commits/7dc7878c037fc3909feb5666638c816606745263

Then I was thinking about an `ExtractorOptions` struct that we could pass in as well, but that also required checking the options in the `extract()` call at runtime. While that's not the end of the world, that's additional check we have to perform but it's a useless check for 99% of the files we scan. To make things a bit worse, the extract() function is called for every line in every file. So there could be a _lot_ of unnecessary checks.

So instead of checking it in this hot path, I created a separate function instead.